dandischema/conf.py

```diff
@@ -0,0 +1,28 @@
+# This file defines the configuration for the DANDI schema
+
+from __future__ import annotations
+
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+
+class Config(BaseSettings):
+    """
+    Configuration for the DANDI schema
+
+    Note
+    ----
+        Since this class is subclass of `pydantic.BaseSettings`, each field of an
+        instance of this class can be populated from an environment variable of the
+        same name prefixed with the prefix defined in `model_config` with the name
+        of the environment variable interpreted case-insensitively.
+        For details, see https://docs.pydantic.dev/latest/concepts/pydantic_settings/
+    """
+
+    model_config = SettingsConfigDict(env_prefix="dandi_schema_")
+
+    nskey: str = "dandi"
+    id_prefix_pattern: str = r"[A-Z]+"
+    """Regex pattern for the prefix of identifiers"""
```

Comment:
please accompany this PR with configs for 

- dandi
- dandi staging (AKA sandbox)
- ember  (for EMBER -- talk to Eric & Sandy)

for `dandi` config, I think you could provide it as one of the CI matrix runs  (or may be we could do for all)

Reply:
I don't think I get this one exactly. With `config.Config`, dandi-schema now can be "vendorized", configured, for a particular vendor, such as dandi or ember. Shouldn't the most direct tests be the testing of validation of a vendorized `Dandiset` models against expected valid metadata instances from respective particular vendors? However, this route may requires a bit maintenance down the road since each time the model changes, we may have to update the metadata instances used in the tests.

Reply:
here is the invocation I did
```
DANDI_SCHEMA_id_pattern=EMBER DANDI_SCHEMA_DATACITE_DOI_ID_PATTERN='(60533|82754)' python -m pytest -s -v dandischema/tests
```

which failed only a few tests... I pushed a commit to potentially fixup one or two tests